Tools/YamlCreate.ps1

```diff
@@ -2576,36 +2628,141 @@ if (($script:Option -eq 'QuickUpdateVersion') -and ($ScriptSettings.SuppressQuic
+        if (!(Test-Path -Path "$FromAppFolder")) {
+          Write-Host -ForegroundColor 'Red' -Object "No manifests found for $OldPackageIdentifier"
+        } else {
+          $manifestsExist = $true
+          Write-Host
+        }
+      } while (!$manifestsExist)
+
+      # Request the new identifier
+      Write-Host -ForegroundColor 'Green' -Object "What is the new package identifier?" -NoNewline
+      $NewPackageIdentifier = Read-PackageIdentifier -PackageIdentifier $null
+      $ToAppFolder = Join-Path $ManifestsFolder -ChildPath $NewPackageIdentifier.ToLower().Chars(0) | Join-Path -ChildPath $NewPackageIdentifier.Replace('.',[IO.Path]::DirectorySeparatorChar)
+      Write-Host
+
+      # Request the new moniker, in case the moniker needs to be updated
+      do {
+        Write-Host -ForegroundColor 'Red' $script:_returnValue.ErrorString()
+        Write-Host -ForegroundColor 'Yellow' -Object '[Optional] Enter the Moniker (friendly name/alias). For example: vscode'
+        if (Test-String -not $NewMoniker -IsNull) { Write-Host -ForegroundColor 'DarkGray' "Old Variable: $NewMoniker" }
+        $NewMoniker = Read-Host -Prompt 'Moniker' | ToLower | TrimString | NoWhitespace
+        if (Test-String $NewMoniker -MaxLength $Patterns.MonikerMaxLength -AllowNull) {
+          $script:_returnValue = [ReturnValue]::Success()
+        } else {
+          $script:_returnValue = [ReturnValue]::LengthError(1, $Patterns.MonikerMaxLength)
+        }
+      } until ($script:_returnValue.StatusCode -eq [ReturnValue]::Success().StatusCode)
+
+      # Get a list of the versions to move
+      $VersionsToMove = @(Get-ChildItem -Path $FromAppFolder | Where-Object {@(Get-ChildItem -Directory -Path $_.FullName).Count -eq 0}).Name
+
+      # Create an array for logging all the branches that were created
+      $BranchesCreated = @()
+
+      foreach ($Version in $VersionsToMove) {
+        Write-Host
+        Write-Host -ForegroundColor Yellow -Object "Moving version $Version [$(1+$BranchesCreated.Count/2)/$($VersionsToMove.Count)]"
+        # Copy the manifests to the new directory
+        $SourceFolder = Join-Path -Path $FromAppFolder -ChildPath $Version
+        $DestinationFolder = Join-Path -Path $ToAppFolder -ChildPath $Version
+        Copy-Item -Path $SourceFolder -Destination $DestinationFolder -Recurse -Force
+        # Rename the files
+        Get-ChildItem -Path $DestinationFolder -Filter "*$OldPackageIdentifier*" -Recurse | ForEach-Object {Rename-Item -Path $_.FullName -NewName $($_.Name -replace [regex]::Escape($OldPackageIdentifier),"$NewPackageIdentifier")}
+        # Update PackageIdentifier in all files
+        Get-ChildItem -Path $DestinationFolder -Filter "*$NewPackageIdentifier*" -Recurse | ForEach-Object  {[System.IO.File]::WriteAllLines($_.FullName, $((Get-Content -Path $_.FullName -Raw) -replace [regex]::Escape($OldPackageIdentifier),"$NewPackageIdentifier"), $Utf8NoBomEncoding)}
```

Comment:
If the manifest contains a newline at the end of the file, `WriteAllLines` will cause it to have 2 ending lines, as seen in this PR 

https://github.com/microsoft/winget-pkgs/compare/master...mdanish-kh:Move-Anydo.Anydo-v5.0.53.4?expand=1

There are couple of workarounds listed here: https://stackoverflow.com/questions/11689337/net-file-writealllines-leaves-empty-line-at-the-end-of-file

Reply:
I think I already fixed this in [cef8a37](https://github.com/microsoft/winget-pkgs/pull/136077/commits/cef8a37b5dcb387aa59dde7497733cabf46e4644). At least when I tried it with Google.Chrome it didn't have any newlines at the end of the file, even though the version in master does have empty lines at the end

https://github.com/microsoft/winget-pkgs/compare/master...Trenly:winget-pkgs:Move-Google.Chrome-v120.0.6099.268-27DDBD?expand=1